src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,69 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    /*
+
+    there is no user with such login in the Storage yet
+    user's login is at least 6 characters
+    user's password is at least 6 characters
+    user's age is at least 18 years old
+
+    */
+    private static final int MIN_USER_LOGIN_SIZE = 6;
+    private static final int MIN_USER_PASSWORD_SIZE = 6;
+    private static final int MIN_USER_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("User cannot be null!");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null!");
+        }
+        if (user.getPassword().isEmpty()) {
+            throw new
+                    RegistrationException("Password cannot be empty!");
+        }
+        if (user.getLogin() == null) {
+            throw new
+                    RegistrationException("Login cannot be null!");
+        }
+        if (user.getLogin().isEmpty()) {
+            throw new
+                    RegistrationException("Login cannot be empty!");
+        }
+        if (user.getAge() == 0) {
+            throw new
+                    RegistrationException("Age must be greater that 0");
+        }
+        if (user.getAge() < 0) {
+            throw new
+                    RegistrationException("Age cannot be negative number!");
```

Comment:
![image](https://github.com/mate-academy/jv-registration-validation/assets/63095184/9cdbe540-4624-4cd1-af93-27367ead37bb)
You already have validation